src/scripts/main.js

```diff
@@ -1,3 +1,194 @@
 'use strict';
 
-// write code here
+const tableHeaders = document.querySelectorAll('th');
+const rowsArray = Array.from(document.querySelectorAll('tbody tr'));
+const sortDirection = {};
+
+tableHeaders.forEach((header, index) => {
+  sortDirection[index] = true;
+
+  header.addEventListener('click', (e) => {
+    const sortedRows = rowsArray.sort((rowA, rowB) => {
+      const cellA = rowA.cells[index].textContent.trim();
+      const cellB = rowB.cells[index].textContent.trim();
+
+      if (index === 4) {
+        const salaryA = parseFloat(cellA.replace(/[$,]/g, ''));
+        const salaryB = parseFloat(cellB.replace(/[$,]/g, ''));
+
+        return sortDirection[index] ? salaryA - salaryB : salaryB - salaryA;
+      } else {
+        return sortDirection[index]
+          ? cellA.localeCompare(cellB)
+          : cellB.localeCompare(cellA);
+      }
+    });
+
+    const tbody = document.querySelector('tbody');
+
+    tbody.innerHTML = '';
+    sortedRows.forEach((row) => tbody.appendChild(row));
+
+    sortDirection[index] = !sortDirection[index];
+  });
+});
+
+const rows = document.querySelectorAll('tbody tr');
+
+rows.forEach((row) => {
+  row.addEventListener('click', () => {
+    rows.forEach((r) => r.classList.remove('active'));
+
+    row.classList.add('active');
+  });
+});
+
+const form = document.createElement('form');
+
+form.classList.add('new-employee-form');
+
+const nameLabel = document.createElement('label');
+
+nameLabel.innerHTML =
+  'Name: <input name="name" type="text" data-qa="name" required>';
+form.appendChild(nameLabel);
+
+const positionLabel = document.createElement('label');
+
+positionLabel.innerHTML =
+  'Position: <input name="position" type="text" data-qa="position">';
+form.appendChild(positionLabel);
+
+const ageLabel = document.createElement('label');
+
+ageLabel.innerHTML =
+  'Age: <input name="age" type="number" data-qa="age" required>';
+form.appendChild(ageLabel);
+
+const salaryLabel = document.createElement('label');
+
+salaryLabel.innerHTML =
+  'Salary: <input name="salary" type="number" data-qa="salary" required>';
+form.appendChild(salaryLabel);
+
+const officeSelectLabel = document.createElement('label');
+
+officeSelectLabel.innerHTML = 'Office: ';
+
+const officeSelect = document.createElement('select');
+
+officeSelect.name = 'office';
+officeSelect.setAttribute('data-qa', 'office');
+
+const cities = [
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+];
+
+cities.forEach((city) => {
+  const option = document.createElement('option');
+
+  option.value = city;
+  option.textContent = city;
+  officeSelect.appendChild(option);
+});
+
+officeSelectLabel.appendChild(officeSelect);
+form.appendChild(officeSelectLabel);
+
+const button = document.createElement('button');
+
+button.type = 'submit';
+button.textContent = 'Save to table';
+form.appendChild(button);
+
+document.body.appendChild(form);
```

Comment:
This form creation code could be simplified and made more readable by using a template literal for the entire form instead of creating and appending each element separately. This will make your code cleaner and easier to understand.